core/test/unit/web/middleware/uncapitalise_spec.js

```diff
@@ -125,17 +125,22 @@ describe('Middleware: uncapitalise', function () {
     });
 
     describe('An API request', function () {
-        ['v0.1', 'v2', 'v3', 'canary', 'v10'].forEach((apiVersion) => {
-            describe(`for ${apiVersion}`, function () {
-                it('does nothing if there are no capitals', function (done) {
+        let apiVersion;
+        describe(`for ${apiVersion}`, function () {
+            beforeEach(function () {
+                apiVersion = ['v0.1', 'v2', 'canary', 'v10'];
+            });
+            it('does nothing if there are no capitals', function (done) {
+                apiVersion.forEach((apiVersion) => {
```

Comment:
@dhuang612 don't think this way of iteration over the `apiVersions` would work in this case (and all the other cases similar to this). The reason being, `done()` will be called during the first iteration of the `it` block and only `v0.1` would be checked. 

After thinking about this problem for a bit I think it's most reasonable just to add an es-lint ignore rule for `no-setup-in-describe` rule with a short note explaining the reason for ignore. Something along the lines:
```
// eslint-disable no-useless-escape
// NOTE: there is no good way to substitute dynamic test setup based on an array of values, see {link to this PR}
```